src/entities.rs

```diff
@@ -25,22 +27,26 @@ impl Entity {
     ///
     /// No particular structure is guaranteed for the returned bits.
     ///
+    /// If bits does not correspond to a possible `NonZeroU64` value returns `None`
+    ///
     /// Useful for storing entity IDs externally, or in conjunction with `Entity::from_bits` and
     /// `World::spawn_at` for easy serialization. Alternatively, consider `id` for more compact
     /// representation.
-    pub fn to_bits(self) -> u64 {
-        u64::from(self.generation) << 32 | u64::from(self.id)
+    pub fn to_bits(self) -> Option<NonZeroU64> {
+        NonZeroU64::new(NonZeroU64::from(self.generation).get() << 32 | u64::from(self.id))
     }
 
-    /// Reconstruct an `Entity` previously destructured with `to_bits`
+    /// Reconstruct an `Entity` previously destructured with `to_bits` if the `bits` is a value
+    /// that can correspond to possible `NonZeroU64`
+    /// 
     ///
     /// Useful for storing entity IDs externally, or in conjunction with `Entity::to_bits` and
     /// `World::spawn_at` for easy serialization.
-    pub fn from_bits(bits: u64) -> Self {
-        Self {
-            generation: (bits >> 32) as u32,
+    pub fn from_bits(bits: u64) -> Option<Self> {
+        NonZeroU32::new((bits >> 32) as u32).map(|_| Self {
+            generation: NonZeroU32::new((bits >> 32) as u32).unwrap(),
```

Comment:
It is redundant to call `NonZeroU32::new` twice. The argument passed in by `map`, which you're currently throwing away with `_`, is desired `NonZeroU32` value already. You could also use the example I gave in the previous review, which instead uses `?`.